### PR TITLE
Masq 119 add mk and encryption at rest

### PR DESCRIPTION
**Pull request by leventdem**
Modifications:

- derivePassphrase returns a protectedMK object which is stored in /profile/protectedMK (not encrypted)
- all values of Masq-profile are encrypted, add a put and get method 
- during a register of a new app, a userAppDEK (Data Encryption Key) is created and stored in app metadata
- the userAppDEK is sent with masqAccessGranted and authorized messages

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "typeface-asap": "0.0.54",
     "uuid": "^3.3.2",
     "webrtc-swarm": "git+https://github.com/QwantResearch/webrtc-swarm#masq",
-    "masq-common": "git+https://github.com/QwantResearch/masq-common"
+    "masq-common": "git+https://github.com/QwantResearch/masq-common#MASQ-118-MK-protection"
```
Comment (G-Ray) [inline package.json]: don't forget to use the master branch now

File: src/library/masq.js [hunk 2]
```diff
@@ -122,8 +158,8 @@ class Masq {
    * Get private profile from hyperdb
    */
   async getProfile (profileId) {
-    this._checkProfile()
-    const profile = (await this.profileDB.getAsync('/')).value
+    // const profile = (await this.profileDB.getAsync('/')).value
```
Comment (G-Ray) [inline src/library/masq.js]: rm comment pls

File: src/library/masq.js [hunk 3]
```diff
@@ -150,7 +186,8 @@ class Masq {
       id: id
     }
 
-    await this.profileDB.putAsync('/', updatedPrivateProfile)
+    await this.put('/', updatedPrivateProfile)
+    // await this.profileDB.putAsync('/', updatedPrivateProfile)
```
Comment (G-Ray) [inline src/library/masq.js]: rm comments

File: src/library/masq.js [hunk 4]
```diff
@@ -69,7 +70,7 @@ class Masq {
       db.on('ready', () => this._startReplicate(db))
     })
 
-    const profile = (await this.profileDB.getAsync('/')).value
+    const profile = await this.get('/')
     return profile
   }
```
Comment (G-Ray) [inline src/library/masq.js:75]: You forgot to clear `this.masterKey` in `closeProfile`

File: package.json (hunk 1 above)
Comment (leventdem) [reply package.json]: done

File: src/library/masq.js (hunk 2 above)
Comment (leventdem) [reply src/library/masq.js]: done

File: src/library/masq.js (hunk 4 above)
Comment (leventdem) [reply src/library/masq.js:75]: Good ! done.

File: src/library/masq.js (hunk 3 above)
Comment (leventdem) [reply src/library/masq.js]: done

File: src/library/masq.js [hunk 5]
```diff
@@ -100,17 +104,47 @@ class Masq {
...
     // Create a DB for this profile
     const db = openOrCreateDB(id)
     await dbReady(db)
-    await db.putAsync('/', privateProfile)
+
+    // we do not use this.encryptAndPut because this value is not encrypted
+    await db.putAsync('/profile/protectedMK', protectedMK)
+    // We do not use this.encryptAndPut method because the master key
+    // is not stored in this.profileDB (only set in openProfile)
+    await put(db, this.masterKey, '/', privateProfile)
```
Comment (G-Ray) [inline src/library/masq.js]: can you store at /profile to avoid the error we discussed in the sync PR please ?

**Pull request closed without merging**